src/node_worker.cc

```diff
@@ -389,14 +399,62 @@ void Worker::New(const FunctionCallbackInfo<Value>& args) {
   }
 
   std::string url;
+  std::shared_ptr<PerIsolateOptions> per_isolate_opts = nullptr;
+
   // Argument might be a string or URL
-  if (args.Length() == 1 && !args[0]->IsNullOrUndefined()) {
+  if (args.Length() > 0 && !args[0]->IsNullOrUndefined()) {
     Utf8Value value(
         args.GetIsolate(),
         args[0]->ToString(env->context()).FromMaybe(v8::Local<v8::String>()));
     url.append(value.out(), value.length());
+
+    if (args.Length() > 1 && args[1]->IsArray()) {
+      v8::Local<v8::Array> array = args[1].As<v8::Array>();
+      // The first argument is reserved for program name, but we don't need it
+      // in workers.
+      std::vector<std::string> exec_argv = {""};
+      uint32_t length = array->Length();
+      for (uint32_t i = 0; i < length; i++) {
+        v8::Local<v8::Value> arg;
+        if (!array->Get(env->context(), i).ToLocal(&arg)) {
+          return;
+        }
+        Utf8Value arg_utf8_value(
+            args.GetIsolate(),
+            arg->ToString(env->context()).FromMaybe(v8::Local<v8::String>()));
```

Comment:
It might be better to also return if `arg->ToString(...)` is empty, similar to the `->Get()` call above?